app/helpers/answers_helper.rb

```diff
@@ -33,12 +33,18 @@ def concat_answer(answer, question)
     elsif question.eql?("support_address")
       answer.values.compact.join(",<br>")
     elsif question.eql?("date_of_birth")
-      Time.zone.local(answer["year"], answer["month"], answer["day"]).strftime("%d/%m/%Y") if answer.present?
+      Time.zone.local(answer["year"], answer["month"], answer["day"]).strftime("%d/%m/%Y") if complete_date?(answer)
     else
       answer.values.compact.join(" ")
     end
   end
 
+  def complete_date?(date)
+    date.present? &&
+      %w(day month year).sort == date.keys.sort &&
```

Comment:
```suggestion
      %w(day month year).each { |required_key| date.key?(required_key) }
```
Nit: might be clearer?

Reply:
Nice!

Reply:
So it turns out if you use `each` and the condition fails, it will return the list of keys, i.e. ["day", "month", "year"], but if you use `all?` instead it works as expected:

`%w(day month year).all? { |required_key| date.key?(required_key) }`